docs/tutorials/particle-filtering.md

```diff
@@ -1,3 +1,564 @@
 # Primer on Particle Filtering with Beluga
 
-Coming soon!
+:::{figure} ../_images/beluga_tutorial.gif
+:alt: Short video of a beluga_tutorial's record.
+:::
+
+## Overview
+
+Modern robotic systems operate in unstructured, unpredictable environments, making robust software crucial to handle diverse situations.
+A key challenge is uncertainty, which occurs when robots lack critical task information. Probabilistic robotics addresses this by
+representing uncertainty explicitly through probability theory, rather than relying on a single best guess.
+
+The robot localization problem is one of the most important ones in mobile robotics, as it is a fundamental requirement for autonomous navigation. The Monte Carlo Localization (MCL) algorithm is one of the most widely used algorithms for map-based localization and has become a key component in many robotic systems [^BelugaPaper]. **TODO: It is necessary to put here the reference, or just include it at the end?**
+
+
+The purpose of this tutorial is to show how to implement a custom {abbr}`MCL (Monte Carlo Localiaztion)` algorithm using **Beluga**,
+motivated by the example given by _Probabilistic Robotics [^ProbRob], page 200, figure 8.11_. **TODO: Can I used the image of the book?**
+
+The code is written using C++17 with the ranges-v3 [^RangesV3] library for range handling along with its respective algorithms.
+
+## Requirements
+
+Before starting the tutorial, first [install Beluga](../getting-started/installation.md).
+
+:::{important}
+I highly recommend reading the [key concepts](../concepts/key-concepts.md) section before diving into this tutorial.
+:::
+
+## Tasks
+
+Based on [key concepts](../concepts/key-concepts.md), this tutorial will show an implementation of the particle filter by performing the recursive update in two steps:
+* [Prediction](#35-prediction)
+* [Update](#36-update)
+
+:::{note}
+We are using the [ROS2](https://docs.ros.org/en/humble/index.html) conventions for package management. **TODO: link to some specific tutorial in ROS2 or just provide the link of ROS2 is fine?**
+:::
+
+## 1 Create the Workspace
+
+Open a new terminal and create the workspace directory:
+
+```bash
+mkdir -p beluga_ws/src
+```
+
+## 2 Create the Package
+
+We are going to create a package inside the `beluga_ws/src` path. Therefore, let's start by creating the necessary directories. In the same terminal, run the following command:
+
+```bash
+mkdir -p beluga_ws/src/beluga_tutorial/src
+```
+
+`beluga_tutorial` should have a `package.xml` and a `CmakeLists.txt` in order to be able to use `colcon` for the compilation process. We will create these files later ([for more details](#4-compile-and-run-the-code)).
+
+## 3 Write the Simulation code
+
+To start, create a `beluga_tutorial/src/main.cpp` file and put the following code inside:
+
+```cpp
```

Comment:
meta: For a first look at the code, I would remove all the comments since they are duplicated below.